packages/react-router/lib/components.tsx

```diff
@@ -254,7 +254,15 @@ export function Navigate({ to, replace, state }: NavigateProps): null {
   );
 
   let navigate = useNavigate();
+
+  // Avoid kicking off multiple navigations during dual-renders in strict mode,
+  // or when components get re-rendered due to global navigation-driven
+  // re-renders (i.e., entering a loading state)
+  let isMountedRef = React.useRef(false);
+
   React.useEffect(() => {
+    if (isMountedRef.current) return;
+    isMountedRef.current = true;
     navigate(to, { replace, state });
   });
```

Comment:
Would it also make sense to add `[]` here?  That doesn't handle the strict mode scenario so we still need the ref, but might make it more explicit that this is only intended to run once?

Reply:
I'd do that for clarity 👍